### PR TITLE
9281 durable orchestrator

### PR DESCRIPTION
**Pull request by turadg**
refs: #9281

## Description

AsyncFlow requires that everything passing the membrane is durable. This makes the facade objects durable to conform.

Following https://github.com/Agoric/agoric-sdk/pull/9529 

### Security Considerations
none

### Scaling Considerations
Exo for each chain and each account

### Documentation Considerations
none

### Testing Considerations
Existing coverage

### Upgrade Considerations
not yet deployed

### CODE REVIEW

File: packages/orchestration/src/exos/local-chain-facade.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        const lcaP = V(localchain).makeAccount();
+        const [lca, address] = await Promise.all([lcaP, V(lcaP).getAddress()]);
+        const { holder: account } = makeLocalOrchestrationAccountKit({
+          account: lca,
+          address: harden({
+            address,
+            chainId: localChainInfo.chainId,
+            addressEncoding: 'bech32',
+          }),
+          // FIXME storage path
```
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/local-chain-facade.js:58]: What ticket capture this? Maybe [vstorage logging](https://github.com/Agoric/agoric-sdk/issues/9066) ?
```suggestion
          // FIXME storage path https://github.com/Agoric/agoric-sdk/issues/9066
```

File: packages/orchestration/src/exos/chain-account-kit.js [hunk 2]
```diff
@@ -169,7 +169,7 @@ export const prepareChainAccountKit = zone =>
         },
         async onClose(_connection, reason) {
           trace(`ICA Channel closed. Reason: ${reason}`);
-          // XXX handle connection closing
+          // FIXME handle connection closing
```
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/chain-account-kit.js:172]:
```suggestion
          // FIXME handle connection closing https://github.com/Agoric/agoric-sdk/issues/9192
```

File: packages/orchestration/src/exos/local-orchestration-account.js [hunk 3]
```diff
@@ -67,18 +58,18 @@ const PUBLIC_TOPICS = {
  * @param {Remote<TimerService>} timerService
  * @param {ChainHub} chainHub
  */
-export const prepareLocalChainAccountKit = (
+export const prepareLocalOrchestrationAccountKit = (
```
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/local-orchestration-account.js:61]: Nice rename 👏

File: packages/orchestration/src/exos/local-orchestration-account.js [hunk 4]
```diff
@@ -207,9 +215,9 @@ export const prepareLocalChainAccountKit = (
          * updater will get a special notification that the account is being
          * transferred.
          */
-        /** @type {LocalChainAccount['deposit']} */
-        async deposit(payment, optAmountShape) {
-          return V(this.state.account).deposit(payment, optAmountShape);
+        /** @type {OrchestrationAccount<any>['deposit']} */
+        async deposit(payment) {
+          await V(this.state.account).deposit(payment);
```
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/local-orchestration-account.js:220]: In a future PR we might consider updating `OrchestrationAccountI['deposit']` to return `Promise<Amount<'nat'>>`. I don't see a reason to swallow that response.

Tangentially - with the revised implementation plan for #9193 only `LocalOrchestrationAccount` will have `.deposit()`.

**Review by 0xpatrickdev — APPROVED** [review]

File: packages/orchestration/src/exos/local-chain-facade.js (hunk 1 above)
Comment (turadg) [reply packages/orchestration/src/exos/local-chain-facade.js:58]: Thanks for looking these up. I expect we'll get to them soon so I'll let this merge as is to save the CI time. Maybe include the links on a next PR.

**Pull request merged**